[CI] Windows release failed

note: LINK : fatal error LNK1181: cannot open input file 'sqlite3.lib'

https://github.com/rooch-network/rooch/actions/runs/6900430147/job/18773561295

--- a/.github/workflows/release_asset.yml
+++ b/.github/workflows/release_asset.yml
@@ -51,7 +51,6 @@ jobs:
         run: |
           echo "C:\ProgramData\chocolatey\lib\SQLite\tools" >> $GITHUB_PATH
           echo "SQLITE3_LIB_DIR=C:\ProgramData\chocolatey\lib\SQLite\tools" >> $GITHUB_ENV
-          echo "SQLITE_DATABASE_URL=C:\test.db" >> $GITHUB_ENV
 
       - name: build for ${{ matrix.platform }}
         uses: actions-rs/cargo@v1

--- a/.github/workflows/release_asset.yml
+++ b/.github/workflows/release_asset.yml
@@ -12,9 +12,9 @@ jobs:
       fail-fast: false
       matrix:
         platform:
-          - ubuntu-latest
-          - ubuntu-22.04
-          - macos-latest
+          # - ubuntu-latest
+          # - ubuntu-22.04
+          # - macos-latest
           - windows-latest
     runs-on: ${{matrix.platform}}
     steps:

--- a/.github/workflows/release_asset.yml
+++ b/.github/workflows/release_asset.yml
@@ -42,7 +42,7 @@ jobs:
         run: |
           choco install sqlite
           cd /D C:\ProgramData\chocolatey\lib\SQLite\tools
-          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+          call "C:\Program Files\Microsoft Visual Studio\*\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
           lib /machine:x64 /def:sqlite3.def /out:sqlite3.lib
 
       - name: Set variables for sqlite (Windows)

--- a/.github/workflows/release_asset.yml
+++ b/.github/workflows/release_asset.yml
@@ -35,6 +35,24 @@ jobs:
         with:
           platform: x64
 
+      #fix https://github.com/rooch-network/rooch/issues/1157  with https://github.com/diesel-rs/diesel/blob/master/.github/workflows/ci.yml
+      - name: Install sqlite (Windows)
+        if: matrix.platform == 'windows-latest'
+        shell: cmd
+        run: |
+          choco install sqlite
+          cd /D C:\ProgramData\chocolatey\lib\SQLite\tools
+          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+          lib /machine:x64 /def:sqlite3.def /out:sqlite3.lib
+
+      - name: Set variables for sqlite (Windows)
+        if: matrix.platform == 'windows-latest'
+        shell: bash
+        run: |
+          echo "C:\ProgramData\chocolatey\lib\SQLite\tools" >> $GITHUB_PATH
+          echo "SQLITE3_LIB_DIR=C:\ProgramData\chocolatey\lib\SQLite\tools" >> $GITHUB_ENV
+          echo "SQLITE_DATABASE_URL=C:\test.db" >> $GITHUB_ENV
+
       - name: build for ${{ matrix.platform }}
         uses: actions-rs/cargo@v1
         with:

--- a/.github/workflows/release_asset.yml
+++ b/.github/workflows/release_asset.yml
@@ -12,9 +12,9 @@ jobs:
       fail-fast: false
       matrix:
         platform:
-          # - ubuntu-latest
-          # - ubuntu-22.04
-          # - macos-latest
+          - ubuntu-latest
+          - ubuntu-22.04
+          - macos-latest
           - windows-2022
     runs-on: ${{matrix.platform}}
     steps:

--- a/.github/workflows/release_asset.yml
+++ b/.github/workflows/release_asset.yml
@@ -15,7 +15,7 @@ jobs:
           # - ubuntu-latest
           # - ubuntu-22.04
           # - macos-latest
-          - windows-latest
+          - windows-2022
     runs-on: ${{matrix.platform}}
     steps:
       - name: Checkout code
@@ -42,7 +42,7 @@ jobs:
         run: |
           choco install sqlite
           cd /D C:\ProgramData\chocolatey\lib\SQLite\tools
-          call "C:\Program Files\Microsoft Visual Studio\*\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
           lib /machine:x64 /def:sqlite3.def /out:sqlite3.lib
 
       - name: Set variables for sqlite (Windows)

--- a/.github/workflows/release_asset.yml
+++ b/.github/workflows/release_asset.yml
@@ -37,7 +37,7 @@ jobs:
 
       #fix https://github.com/rooch-network/rooch/issues/1157  with https://github.com/diesel-rs/diesel/blob/master/.github/workflows/ci.yml
       - name: Install sqlite (Windows)
-        if: matrix.platform == 'windows-latest'
+        if: matrix.platform == 'windows-2022'
         shell: cmd
         run: |
           choco install sqlite
@@ -46,7 +46,7 @@ jobs:
           lib /machine:x64 /def:sqlite3.def /out:sqlite3.lib
 
       - name: Set variables for sqlite (Windows)
-        if: matrix.platform == 'windows-latest'
+        if: matrix.platform == 'windows-2022'
         shell: bash
         run: |
           echo "C:\ProgramData\chocolatey\lib\SQLite\tools" >> $GITHUB_PATH

--- a/scripts/release.sh
+++ b/scripts/release.sh
@@ -6,7 +6,7 @@ rm -rf rooch-artifacts/*
 mkdir -p rooch-artifacts/
 cp -v target/release/rooch rooch-artifacts/
 cp -v README.md rooch-artifacts/
-if [ "$1" == "windows-latest" ]; then
+if [ "$1" == "windows-2022" ]; then
   7z a -r rooch-$1.zip rooch-artifacts
 else
   zip -r rooch-$1.zip rooch-artifacts

